src/http/server.clj

```diff
@@ -293,6 +294,29 @@
          (log/infof "\"%s %s HTTP/1.1\" %s %s" method'  uri status len))
        response))))
 
+(defn password-prompt! []
+  (println "Enter password: ")
+  (let [pw (-> (System/console) (.readPassword) (String.))]
+    (if (empty? pw)
+      (do (println "Password cannot be empty!")
+          (recur))
+      pw)))
+
+(defn maybe-inject-auth
+  [auth handler]
+  (if (some? auth)
+    (fn [req]
+      (handler (assoc-in req [:headers "authorization"] auth)))
```

Comment:
Sorry, I didn't spot this at first glance. You need to check the auth here, not just assoc into the request. You need to ensure that auth header is present, otherwise, reply with 401 status code.

Reply:
I understand all your other comments but I am not sure I follow on this one. Wouldn’t auth headers only be present if the flag is set at the command line? What should I be checking for prior to associng them into the request map / how would they be present otherwise? I’ll make the rest of the changes over the next few days - in Vietnam till Saturday so I’m sporadically checking.

Reply:
When you working with basic auth, the flow looks like this:

1. The client sends a request (e.g. browser sends GET when you open the page in the browser)
2. The server checks header "Authorization" (should be something like "Basic <...>")
     2.1 If not provided, the server must respond with `401` status code and non-empty  "WWW-Authenticate" header
     2.2 If presented, the server must validate user/pass and either go to 2.1 or respond with `200 OK`

Reply:
Nicer looking schema: https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication#The_general_HTTP_authentication_framework

Reply:
Thanks, I've read through the link, but the part I don't understand is in what situation this flow would need to occur.  Right now there are two options AFAICT:

1. `--auth` is not set at CLI, in which case no auth headers are present and no credentials need to be passed, the request will proceed unmodified.
2. `--auth` is present, either with or without pw (prompting if not present), and we send those credentials through by associng them in.

In this flow, _would authorization headers ever be present to check if we have not explicitly set them via CLI (e.g. your pt 2 above)_?  Or is it suppose to be that auth is required by default for every resource, and I should reject and trigger this flow when they are not present (in which case the flag isn't really optional, in that every request will 401 without the flag set)?

Sorry if this is belaboring the point - but I'm a bit unsure how to proceed.

Reply:
1. CLI flag not set = the server works as it did previously.
2. CLI flag set and the request does not contain "Authorization" header = 401
3. CLI flag set and the request contains "Authorization" header but user/password mismatched with what was set with CLI flag = 401
4. Otherwise = proceed to a normal execution flow

Reply:
Thanks, I get it now.  I think my confusion is that I don't understand where the auth headers in `3.` are coming from if not CLI.  I'll work on this later tonight/tomm.  BTW - I appreciate you providing an entry point for working on this project and hope to continue once I get this one done.

Reply:
The header in 3 comes from the client which is performing the request. If you open it in a browser, the browser will automatically prompt you to input username and password. Or you can do the same using `curl`, `httpie`, `wget` or any other tool out there.